include/ofi_indexer.h

```diff
@@ -60,7 +60,7 @@ struct ofi_idx_entry {
 	int   next;
 };
 
-#define OFI_IDX_INDEX_BITS 16
+#define OFI_IDX_INDEX_BITS 20
```

Comment:
This will increase the default size of the indexer from an array of 64 pointers to an array of 1024 pointers.  Are we okay with that increase?  Looking at the indexer use in the code, this should be okay.

Reply:
I'm ok with it. The indexer seems to add memory as necessary, so in the worst case scenario (unused space), it adds an additional 7680 bytes per process as overhead, regardless of usage. It also allows the indexer to grow (maximum usage) from 530K per process to 8.2MB per process

Reply:
With 128 processes per node:

10% filled (before): 6.8 MB total, ~0.05MB per process
10% filled (after): 108MB, ~0.8MB per process
100% filled (before): 67 MB, ~.5MB per process. 
100% filled (after): 1.07GB, ~8.2 MB per process

Reply:
Whoa - something doesn't look right here.  This increase should only add a few KB per node.

Reply:
err, per process

Reply:
Right... to the base memory cost. If the indexer were filled to the maximum, then it would consume more because there is more space that could be allocated. 

To the base memory cost (the cost of the 64 or 1024 array locations), it would only be a few KB per process.

Reply:
Okay, I think the issue is that the comparisons above are misleading.  10% filled after the change is significantly more entries added to the index than 10% before the change.  It would be better to show the consumption for X entries total, which I would expect to be the same for any reasonable value of X.

Reply:
I was actually trying to show it a little better... apples to apples. 

working on it

Reply:
With 128 processes per node:
6.4K entries (10% of indexer) [before]: 6.8 MB total, 52KB per process
6.4K entries (0.625 % of indexer) [after]: 7.8MB, 60KB per process

64K entries (100% of original) filled (before): 67 MB, 524K per process. 
64K entries (6.25% of indexer) filled (after): 68.1MB, 532K per process

max entries (before): 67 MB, ~.5MB per process. 
max entries (after): 1.07GB, 8MB per process

Reply:
Agreed though. Initial numbers were a misleading :) Sorry about that.

Reply:
Hmm... an extra MB per process with this change is a lot.  What a I missing?  Why isn't this change adding about 8-16KB per process?

Reply:
I didn't read your data correctly.  This is only about 8KB per process increase.  Thanks